components/function-controller/internal/controllers/serverless/fsm.go

```diff
@@ -130,7 +130,10 @@ func buildGenericStatusUpdateStateFn(condition serverlessv1alpha2.Condition, rep
 
 func (m *reconciler) populateStatusFromSystemState(status *serverlessv1alpha2.FunctionStatus, s *systemState) error {
 	status.Runtime = s.instance.Spec.Runtime
-	status.RuntimeImageOverride = s.instance.Spec.RuntimeImageOverride
+	status.RuntimeImage = s.instance.Status.RuntimeImage
+	if s.instance.Spec.RuntimeImageOverride != "" {
```

Comment:
We can remove this `if`, because:
if s.instance.Spec.RuntimeImageOverride is not empty, it will be saved, but by default `status.RuntimeImage` is empty.

Reply:
Removed